src/engine/controls/loopingcontrol.cpp

```diff
@@ -1078,13 +1078,28 @@ void LoopingControl::slotBeatLoop(double beats, bool keepStartPoint, bool enable
                 //
                 // If we press 1/2 beatloop we want loop from 50% to 100%,
                 // If I press 1/4 beatloop, we want loop from 50% to 75% etc
-                double beat_len = nextBeat - prevBeat;
-                double loops_per_beat = 1.0 / beats;
-                double beat_pos = currentSample - prevBeat;
-                int beat_frac =
-                        static_cast<int>(floor((beat_pos / beat_len) *
-                                                loops_per_beat));
-                newloopSamples.start = prevBeat + beat_len / loops_per_beat * beat_frac;
+                double beatLength = nextBeat - prevBeat;
+                double beatPos = currentSample - prevBeat;
+
+                // find the two closest beat fractions and place the new loop start to the closer one
+                double fractionBeatLength = beatLength * beats;
+                double previousFractionBeat = prevBeat + (floor(beatPos / fractionBeatLength)) * fractionBeatLength;
```

Comment:
The () around is redundant. You can use it to emphasize that * is done first or move the + prevBeat to the end.